lib/rubocop/cop/lint/next_without_accumulator.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # Don't omit the accumulator when calling `next` in a `reduce` block.
+      #
+      # @example
+      #   # bad
+      #   result = (1..4).reduce(0) do |acc, i|
+      #     next if i.odd?
+      #     acc + i
+      #   end
+      #
+      #   # good
+      #   result = (1..4).reduce(0) do |acc, i|
+      #     next acc if i.odd?
+      #     acc + i
+      #   end
+      class NextWithoutAccumulator < Cop
+        MSG = 'Use `next` with an accumulator argument in a `reduce`.'.freeze
+        METHODS = [:inject, :reduce]
+
+        def on_block(node)
+          method, _args, body = *node
```

Comment:
Lines 25-34 can be compressed into one of our "node patterns". The pattern you need should be: `(block (send _ {:reduce :inject} !sym) _ begin)`